src/Http/Routing/src/RouteEndpointBuilder.cs

```diff
@@ -30,10 +30,12 @@ public sealed class RouteEndpointBuilder : EndpointBuilder
     /// <param name="routePattern">The <see cref="RoutePattern"/> to use in URL matching.</param>
     /// <param name="order">The order assigned to the endpoint.</param>
     public RouteEndpointBuilder(
-       RequestDelegate requestDelegate,
+       RequestDelegate? requestDelegate,
```

Comment:
I think this was primarily done because we pass a potentially null `RequestDelegate` already:
https://github.com/dotnet/aspnetcore/blob/318e16b6ac3a9644b266a463c8db47d6a24cad0d/src/Http/Routing/src/EndpointDataSource.cs#L57
EndpointDataSource should probably be updated to not use `!` now.

Reply:
When we approved this in #43000, we discussed leaving it annotated as non-nullable, but we felt uncomfortable actually throwing an ArgumentNullException if it was null after not doing so for so long. The EndpointDataSource code shows how that kind of thing could break working code. And since `RouteEndpointBuilder.RequestDelegate` is nullable anyway, there's really no downside to just fixing the nullability annotation here in the RouteEndpointBuilder constructor.

Good catch about the unneeded `!` in `EndpointDataSource.GetGroupedEndpoints`. I'll remove it.